Sources/AppStoreServerLibrary/SignedDataVerifier.swift

```diff
@@ -148,7 +149,23 @@ public struct SignedDataVerifier {
         return appTransactionResult
     }
     
-    private func decodeSignedData<T: DecodedSignedData>(signedData: String, type: T.Type) async -> VerificationResult<T> where T : Decodable {
-        return await chainVerifier.verify(signedData: signedData, type: type, onlineVerification: self.enableOnlineChecks, environment: self.environment)
+    private func decodeSignedData<T: DecodedSignedData & JWTPayload>(signedData: String, type: T.Type) async -> VerificationResult<T> where T : Decodable {
+        await chainVerifier.verify(signedData: signedData, type: type, onlineVerification: self.enableOnlineChecks, environment: self.environment)
     }
 }
+
+extension AppTransaction: JWTPayload {
```

Comment:
Is it possible to extend or modify DecodedSignedData in some way to keep this more flexible?